src/ol/source/stamensource.js

```diff
@@ -88,36 +88,78 @@ ol.source.StamenProviderConfig = {
  */
 ol.source.Stamen = function(options) {
 
-  var i = options.layer.indexOf('-');
-  var provider = i == -1 ? options.layer : options.layer.slice(0, i);
-  goog.asserts.assert(provider in ol.source.StamenProviderConfig,
-      'known provider configured');
-  var providerConfig = ol.source.StamenProviderConfig[provider];
+  /**
+   * url
+   * @type {string}
+   * @private
+   */
 
-  goog.asserts.assert(options.layer in ol.source.StamenLayerConfig,
-      'known layer configured');
-  var layerConfig = ol.source.StamenLayerConfig[options.layer];
+  this.rootUrl_ = goog.isDef(options.url) ? options.url :
+      'https://stamen-tiles-{a-d}.a.ssl.fastly.net/';
```

Comment:
Indeed, but maybe should we also rename the option as "rootUrl" to be more consistent? If url option is needed, the end user can still use the generic ol.source.XYZ.